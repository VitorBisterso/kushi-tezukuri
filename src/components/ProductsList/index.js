import React from 'react'
import PropTypes from 'prop-types'

import ProductCard from '../ProductCard'

/* TODO: Use firebase images */
import defaultImage from '../../assets/img/defaultImage.jpg'

const ProductsList = ({ products }) =>
  products &&
  products.map(product => {
    const { id, name, price } = product

    return (
      <ProductCard
        key={id}
        title={name}
        price={price}
        image={defaultImage}
        /* TODO: Check if user is logged */
        isAdminPage={false}
      />
    )
  })

ProductsList.propTypes = {
  products: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default ProductsList
