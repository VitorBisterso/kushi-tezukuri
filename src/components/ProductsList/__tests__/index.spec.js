import React from 'react'
import { shallow } from 'enzyme'

import ProductsList from '../index'

describe('ProductsList', () => {
  it('should render correctly', () => {
    const products = [{ id: 1, name: 'Test name', price: 10 }]
    const component = shallow(<ProductsList products={products} />)

    expect(component).toMatchSnapshot()
  })
})
