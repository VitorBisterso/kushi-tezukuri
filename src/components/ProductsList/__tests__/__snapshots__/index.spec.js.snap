// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList should render correctly 1`] = `
<ProductCard
  image=""
  isAdminPage={false}
  key="1"
  price={10}
  title="Test name"
/>
`;
